contrib/LabelsToTags.lua

```diff
@@ -48,7 +48,12 @@
 ]]
 
 local darktable = require("darktable")
-darktable.configuration.check_version(...,{3,0,0},{4,0,0},{5,0,0})
+local du = require "lib/dtutils"
+
+if not du.check_min_api_version("3.0.0") then
+  darktable.print("ERROR:LabelsToTags failed to load.  Lua API version 3.0.0 or later required.")
```

Comment:
I have two comments:
1. I don't like the double  space here: "load.  Lua"
2. Have you tried to generate the error massage in du.check_min_api_version()? Perhaps extend the method with two parameters "showErrorMessage" and "scriptName". What do you think?

Reply:
Ideally I could exit the routine from the function call if the API isn't compatible.  I tried a couple of things.  Did you know that calling os.exit() kills darktable?  I also tried return -2 to try and pop the stack, but that didn't seem to have an effect.  So, at present, I'm stuck with returning true or false and then the caller has to act on it accordingly.  If I generate the error message in the function then the caller fails to act on it (by doing a return) then the function will continue to execute until it crashes on a part of the API that's missing.  I'll see if I can figure out a way to exit the caller from the version check.  If I can, then I'll move the error message.  Or, I can move the message and we can enforce the syntax on anything we include in the lua-scripts.  We could just code it as 

if not check_min_api_version(<version>,<script>) return end

As for the double spaces after the period, my typing teacher beat it into me along with too many years of using a typewriter.  I don't care either way, but muscle memory often wins out.  If we keep the error messages this way, I'll fix it with a perl 1 liner...  :-)